Add support for managing Mikrotik users

dpkingston opened the pull request:
We add support for managing Mikrotik users using the same list of users that was previously used for roles/users which is really mis-named (It really just manages Linux users). We move the relevant file of organization specific information to a more accessible location so the same list of users is available to all roles.  Molecule tests are updated to test the new functionality and full tests pass for all cases. Still to be done is documentation enhancement to explain where to put org/site specific information in topmost README.md.

dpkingston committed:
Don't try to update users on anycast "hosts".
There is no guarantee what host you will wind up connecting to if
multiple hosts provide the service.  Use the hosts specific identity
and address instead.

dpkingston committed:
Commit routeros testing support in a working state.
There is certainly some fine tuning to be done on this but the
ansible + molecule + vagrant support is wired up and working and
successfully processes testing against ROS6 and ROS7.

dpkingston committed:
Additional files needed for initial routeros testing support.

dpkingston committed:
Update reference to README for role routeros_common

dpkingston committed:
Add more description of the testing strategy for routeros_common.

dpkingston committed:
Made program portable to other users by using $HOME reference.

dpkingston committed:
Add comment about need to stay in sync with Vagrant configuration.

dpkingston committed:
This makes the choice of file explicit at the minor cost of some repetition.
It has the advantage of also being more flexible if more segregation of config
is needed.

dpkingston committed:
Remove unnecessary vars.

dpkingston committed:
Merge branch 'master' into update-users-role

dpkingston committed:
Fixed whitespace issue.

dpkingston committed:
Add support for managing users in RouterOS.
To avoid having users lists in more than one place, the vars file from roles/users
has been moved to inventories/psdr/group_vars/owner_HamWAN.yml, where it is available
to all as global information. molecule_vars is provided to provide these variables
for the molecule testing. TODO: provide more documenation.

dpkingston commented:
PS. Something weird happened with syncing this branch with master so there seem to be a bunch of commits listed here that are already applied as they don't show up in the diff.

dpkingston committed:
Add documentation on configuring these ansible modules.

gerner commented on roles/routeros_common/tasks/manage_users.yml:
```diff
@@ -0,0 +1,102 @@
+# Users present and in the delete_users list are delete.
+# For now, we leave extra users as is.
+
+    - name: setup expected users
```
maybe I'm demonstrating my lack of ansible or yaml expertise here, but is there a reason that this whole file starts indented 3 spaces?

dpkingston replied:
No good reason.  Largely he result of some cut/paste and not enough cleanup.  Fixed in next commit.

gerner commented on roles/routeros_common/tasks/manage_users.yml:
```diff
@@ -0,0 +1,102 @@
+    - name: setup expected users
+      set_fact:
+        expected_users: "{{ users_admin + users_user + ['admin'] }}"
+        delete_users: ['DeleteMe1', 'DeleteMe2']
```
this list looks like test code, so I'm surprised to see it outside the molecule tree.

dpkingston replied:
I have removed the hard-coded list and added a new user list for deletions (users_delete).

gerner commented on roles/routeros_common/tasks/manage_users.yml:
```diff
@@ -0,0 +1,102 @@
+
+    - name: Find extra users
+      set_fact:
+        extra_users: >-
```
just want to verify my understanding of the desired behavior:
* ensure we have at least all the users in `expected_users` (which is the union of `users_admin` and `users_user`)
* remove any existing users in `deleted_users` (which is a hardcoded pair of users "DeleteMe1" and "DeleteMe2"?)
* display any other existing users not in `expected_users`, but leave them in place
* there's no distinction between users listed in `users_admin` vs `users_user` (at least for now)

is that correct?

dpkingston replied:
Basically yes.  I should change deleted users to something real or an empty list.  Will do that in the next commit.
I do treat the monitoring user specially.  I should use users_user to do that. Let's see if I can suss that out.

dpkingston replied:
Found and tested a fix for hardcoded monitoring user.  I now use users_user membership.

gerner reviewed (changes requested):
I'm surprised by a couple of things, but that might be my lack of expertise.

dpkingston committed:
Remove hard-coded deletion list. Add parameter for users to delete (users_delete).
Remove hardcoded handling of monitoring users.  Use membership in users_user instead.

gerner reviewed (approved):
assuming you've done some testing on this, it all looks ok to me.

dpkingston commented:
Yes, all the molecule tests pass.
Also ran in --check mode against a real host.

-Doug-


On Mon, Apr 14, 2025 at 12:08 PM Nick Gerner ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> assuming you've done some testing on this, it all looks ok to me.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/HamWAN/infrastructure-configs/pull/27#pullrequestreview-2765404137>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABLQD22RRJUJSETJAHBUNTL2ZQBRTAVCNFSM6AAAAAB2HSDDYCVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDONRVGQYDIMJTG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
